neural_net_basics/neural-network.md

```diff
@@ -0,0 +1,137 @@
+* $b$ is the bias
+* $n$ is the number of inputs
+
+
+![png](/neural_net_basics//assets//neuron.png)
+
+## Why is an activation function important?
+A neuron, with a single input, is a linear function, $y = ax+b$, and as such can only form simple decision boundaries. An increase in the number of inputs and neurons in the neural network will still result in a linear function. This is because multiple combined linear functions always results in a linear function. 
+
+To resolve this issue, and allow the network to find complex decision boundaries, one that wiggles around separating multiple classes, we need to use activation functions, they add non-linearity.
+
+### Step Function
+![png](/neural_net_basics//assets//step.png) 
+$$ f(x) =\begin{cases}0 & x < 0\\1 & x  \geq  0\end{cases} $$
+
+The original activation function, used in the original perceptron in the 1950's. This function converts the output of the summation function into a binary value.
+
+Due to the nature of the step in the function, changes to the weights and bias make no difference until the output crosses the step boundary, at which point it flips the binary value. Due to a very small change to the weight having such a dramatic effect, it is hard to train the neural network.
+
+### Logistic Function a.k.a Sigmoid function
+![png](/neural_net_basics//assets//sigmoid.png)
+$$ f(x) = \frac{1}{1 + e^{-x}}$$
+The Logistic function, a variant of the sigmoid function, initially replaced the step function. Like the Step function it limits the output of a neuron between 0 and 1, but doesn't have the same issues around the zero mark. This makes it easier to train as a change to the weights and biases will always have an effect, except at the extremes.
+
+### Rectified Linear Unit (ReLU)
+![png](/neural_net_basics//assets//relu.png)
+$$ f(x) = max(0,x) $$
+The Rectified Linear  (ReLU) has become the most popular and in most cases the default activation function to use for deep neural networks. It takes the maximum value of either 0 or the input, essentially putting any negative input value to 0 immediately.
+
+### SoftMax
+$$ \delta(z) = \frac{e^{z_{i}}}{\sum_{j=1}^{K}e^{z_{j}}} $$
+The SoftMax Activation function is used for the final output layer for classification neural networks. It converts the output values into a probability distribution over the output classes. Each output neuron will represent the probability the input is of that class.
+
+## Architecture of a Neural Network
+
+In neural networks, the neurons can be arranged in a variety of ways. For the basic neural network the neurons are arranged into several distinct layers. 
+
+There are three layer categories
+
+1. Input Layer
+2. Hidden Layers
+3. Output Layer
+
+### 1. Input Layer
+The input layer is purely the values of the input data, no calculations take place. There is a neuron for each input value, for image classification neural networks, for grey scale images, there would be a node for each pixel. 
```

Comment:
What's the difference between a node and a neuron? If they are distinct, it would be good to describe the difference. If they are synonyms, it aids understanding to always use the canonical term.